src/customers/customers.service.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Customer } from './customers.entity';
+import { CreateCustomerDto } from '../dto/create-customer.dto';
+import { UpdateCustomerDto } from '../dto/update-customer.dto';
+
+@Injectable()
+export class CustomersService {
+  constructor(
+    @InjectRepository(Customer)
+    private readonly customersRepository: Repository<Customer>,
+  ) {}
+
+  notFound = (error) => {
+    throw new NotFoundException(`Customer not found`, { cause: error });
+  };
```

Comment:
Awesome idea!
Maybe we could extract this and include other errors to separate file too so we could reuse all errors in any service/controller?